src/Protego.t.sol

```diff
@@ -65,337 +46,239 @@ interface DSTokenLike {
     function transferFrom(address, address, uint256) external returns (bool);
 }
 
-contract Stranger {
-    function plot(DSPauseTester pause, address usr, bytes32 tag, bytes memory fax, uint eta) public {
-        pause.plot(usr, tag, fax, eta);
-    }
-    function drop(DSPauseTester pause, address usr, bytes32 tag, bytes memory fax, uint eta) public {
-        pause.drop(usr, tag, fax, eta);
-    }
-    function exec(DSPauseTester pause, address usr, bytes32 tag, bytes memory fax, uint eta)
-        public returns (bytes memory)
-    {
-        return pause.exec(usr, tag, fax, eta);
-    }
-}
-
 struct BadSpells {
     address badSpell;
     address action;
     bytes32 tag;
-    bytes   sig;
+    bytes sig;
     uint256 eta;
 }
 
-contract BadAction {
+contract MaliciousSpell {
```

Comment:
I'm gonna rearrange the contracts the way you're suggesting, however I don't get this part:

> Make NonConformantSpell define `action`, `tag`, `sig`, `eta` in way that is non-conformant / malicious

It's already doing that. `MaliciousSpell` allows anyone to `plot` any `action` + `tag` + `sig` + `eta` combination.

Reply:
I meant that the `NonConformantSpell` can actually try to be malicious: it can define `action`, `tag`, `sig`, `eta` not as public _variables_ [like ConformantSpell does](https://github.com/makerdao/protego/blob/9eeb52c046e98fa2f377d480e403fccf13662872/src/test/ConformingSpell.sol#L41-L45), but as public _methods_ with custom malicious logic. For example: `tag` can return a different value when called by the pause compared to when called by protego. Meaning that it would still be conformant interface-wise, but not conformant logic-wise: making it callable via `deploy(DssSpellLike _spell)(address)` which would then deploy incorrect DropSpell.

Reply:
A scenario like that cannot be technically prevented from a code perspective, it is more like a process issue.

I understand what you are saying, but IMO it does not help to "prove" the correctness of the implementation, which is the main goal of the tests here.
It should be clear that for non-conforming spells the version with the parameters should be used. This applies even if a non-conforming spell looks like a conforming one.

Reply:
> cannot be technically prevented from a code perspective

Design-wise, I think if we don't even have `deploy(DsSpellLike _spell)(address)` and `drop(DsSpellLike _spell)` methods and instead have a script that parses required data from the original `plot` transaction it would be decrease probability of mistakes. Of course someone could still just trust the spell, but it would be less convenient.

> IMO it does not help to "prove" the correctness of the implementation, which is the main goal of the tests here.

I understand that theoretically it should not make a difference, but I personally usually rather make a test more realistic than try to imagine all edge cases that less realistic test leaves out. So I can agree that it's an opinionated suggestion

Reply:
Hmm, I think I get your point. If we're talking about a conforming spell, it most likely was written by one of the teams in the spell roster, so they could easily assist governance facilitators figuring out the parameters. Maybe it makes sense to reduce the interaction surface by removing the redundant methods.

There are some things to consider, however:

> [...] instead have a script that parses required data from the original plot transaction it would be decrease probability of mistakes [...]

Who would be the end users of such script?
We want Protego to be usable regardless of the availability of engineering teams.
I guess we most likely need a UI instead of a script.

Reply:
> Who would be the end users of such script?
We want Protego to be usable regardless of the availability of engineering teams.
I guess we most likely need a UI instead of a script.

I agree that the end goal is enabling non-engineers to run it. But my other point is to include the usage into the scope of the audit. Also, nothing stops you from writing a simple js script that is then directly importable as a dependency into the UI

Reply:
Agreed. However, could we do that in a follow-up PR?